Tests/Common/Utils/XCTestCase+SupportedProtocols.swift

```diff
@@ -25,4 +25,22 @@ extension XCTestCase {
             sleep(transportProtocol == last ? 0 : interval)
         }
     }
+
+    @available(iOS 13.0, *)
+    func testSupportedProtocols(
+                description: String,
+                timeout: Double = 40.0,
```

Comment:
Couple of questions on this part - GRPC and HTTP currently have different timeouts...only thing I think we need for the tests is that the test timeout is slightly longer than the protocol timeout so the calls can legitimately fail prior to the test timeout for expected error conditions but I don't see where timeout or description are actually used here...The test timeout being the greater of the two values should be fine as the protocol timeouts will occur first. If we could base the default on the greater of the two values plus a slight increment, that'd likely be best.

For the interval delay, do we actually need it?  Will add +10s  for every protocol related test, right?

Reply:
10s was existing configuration from the non-async version of this function. I think 10s is the right amount to make sure we don't run into race conditions. For local-dev, you can use the ExampleHTTP project and there will be no interval because theres only one protocol.

Reply:
Updated with timeout logic